front/lib/api/assistant/agent.ts

```diff
@@ -1393,6 +1386,24 @@ export class TokenEmitter {
     yield* this.flushTokens();
   }
 
+  async parseContents(
```

Comment:
this is maybe a bit of a hack since we're creating a useless async generator and a bunch of useless events (that are swallowed). So there's a little bit of JS overhead. 
But this allows having one single implementation for the whole parsing logic which makes everything much simpler. 
It's "fake async" since it doesn't actually do any IO / network